Sources/Contentful/Client.swift

```diff
@@ -121,10 +121,15 @@ open class Client {
         }
 
         self.persistenceIntegration = persistenceIntegration
-        let contentfulHTTPHeaders = [
+        var contentfulHTTPHeaders: [AnyHashable: Any] = [
             "Authorization": "Bearer \(accessToken)",
             "X-Contentful-User-Agent": clientConfiguration.userAgentString(with: persistenceIntegration)
         ]
+        if let additionalHeaders = sessionConfiguration.httpAdditionalHeaders {
+            contentfulHTTPHeaders.merge(additionalHeaders) { (currentPair, _) in currentPair }
+        }
+        sessionConfiguration.httpAdditionalHeaders = contentfulHTTPHeaders
```

Comment:
Merging http headers from session configuration passed by user.

Reply:
Basically the current code overwrite `httpAdditionalHeaders` with a hard coded `contentfulHTTPHeaders` within the local scope, and we suspect this is not the intended behavior. Please check if this change is acceptable. Thanks!